explainers/subresource-loading.md

```diff
@@ -0,0 +1,102 @@
+  tools but configuring them to work in an optimal way is tough, and sometimes
+  also conflicts with new requirements like dynamic bundling (e.g. small edit
+  with tree shaking could invalidate everything).
+
+- With JS bundlers, parsing and execution needs to wait for the full bytes to
+  come. Ideally loading multiple subresources should be able to utilize full
+  streaming and parallelization, while it's not possible if all resources are
+  bundled as one javascript. (For JS modules execution still needs to be waited
+  for the entire tree due to the current
+  [deterministic execution model](https://docs.google.com/document/d/1MJK0zigKbH4WFCKcHsWwFAzpU_DZppEAOpYJlIW7M7E/edit#heading=h.5652gd5ks5id))
+
+- Related issues: [#411](https://github.com/WICG/webpackage/issues/411),
+  [#526](https://github.com/WICG/webpackage/issues/526)
+
+## Proposal
+
+We propose a new `link rel=”webbundle”` notion [**1**] that lets the browser
+fetch bundled resources and attach them to the page. Such as:
+
+```html
+<link
+  rel="webbundle"
+  href="https://example.com/dir/subresources.wbn"
+  scope="https://example.org/scope/"
+/>
+```
+
+Both the `href` and the `scope` need to be the same-origin.
+
+If the main document has such a `link` element, a browser must do:
+
+- A browser must fetch the specified Web Bundle, `subresources.wbn`.
+
+- All the subresources whose URLs are under the scope
+  `https://example.org/scope/` must [**2**] be loaded from the specified Web
+  Bundle.
+
+[**1**] The syntax is tentative. The API surface is not so important for us yet.
+Any alternative ideas are welcome. This is basically about mapping a set of
+resources to a particular bundled resource. Something like Fetch maps proposal
+might be able to extend to this case.
+
+[**2**] The exact behavior is TBD. There are several possible behaviors when the
+bundle does not contain the subresource; e.g. just fail, fallback to the
+network, or allow users to specify its behavior with a `link` element's
+attribute.
+
+## Example
+
+### The bundle
+
+Suppose that the bundle, `subresources.wbn`, includes the following resources:
+
+```
+- https://example.org/scope/a.js (which depends on ./b.js)
+- https://example.org/scope/b.js
+- https://example.org/scope/c.png
+- … (omitted)
+```
+
+### The main document
+
+```html
+<link rel="webbundle"
+    href="https://example.com/dir/subresources.wbn"
+    scope="https://example.org/scope/"
+/>
+
+<script type=”module” src=”https://example.com/scope/a.js”></script>
+<img src=https://example.org/scope/c.png>
+```
+
+Then, a browser must fetch the bundle, `subresources.wbn`, and load
+subresources, `a.js`, `b.js`, and `c.png`, from the bundle.
+
+## Subsequent loading and Caching
+
+On the subsequent navigations for the same page/origin, it’s desirable that the
+resources that are already fetched in the bundle shouldn't be refetched. We're
+thinking about two possibilities here:
+
+1. Embed the details of version / cached resources in the URL
```

Comment:
Could you expand this a bit? I'm not sure how this enables resource subsetting.

Reply:
@kinu 
Do you have any opinions? I guess we don't have a concrete idea yet.

Reply:
Can we address this in a follow-up commit? We can adress that later, I think.

Reply:
sgtm.

Reply:
Yeah... I think I was thinking *something* like: Server can maintain versioned bundles, and client can send a request with some URL parameters that indicates the version of the bundle they already have in cache (so that the server can identify the updated resources and send them back).  This is mostly about updating-fetch for same bundle cases though, and haven't really thought about how to do for similarly looking bundle on different pages cases.

Reply:
@kinu Thanks! I appreciate that! That's helpful. I think we can update the explainer with the more details once we can explore this area. Let me close this as resolved for now.